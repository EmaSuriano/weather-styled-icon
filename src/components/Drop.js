import styled, { keyframes } from 'styled-components';
import PropTypes from 'prop-types';
import { ThemePropType, defaultTheme } from '../constants';

export const DropAnimation = ({ theme }) => keyframes`
0% {
  background: ${theme.dropsColor};
  box-shadow: 0.625em 0.875em 0 -0.125em rgba(255, 255, 255, 0.2),
    -0.875em 1.125em 0 -0.125em rgba(255, 255, 255, 0.2),
    -1.375em -0.125em 0 ${theme.dropsColor};
}
25% {
  box-shadow: 0.625em 0.875em 0 -0.125em rgba(255, 255, 255, 0.2),
    -0.875em 1.125em 0 -0.125em ${theme.dropsColor},
    -1.375em -0.125em 0 rgba(255, 255, 255, 0.2);
}
50% {
  background: rgba(255, 255, 255, 0.3);
  box-shadow: 0.625em 0.875em 0 -0.125em ${theme.dropsColor},
    -0.875em 1.125em 0 -0.125em rgba(255, 255, 255, 0.2),
    -1.375em -0.125em 0 rgba(255, 255, 255, 0.2);
}
100% {
  box-shadow: 0.625em 0.875em 0 -0.125em rgba(255, 255, 255, 0.2),
    -0.875em 1.125em 0 -0.125em rgba(255, 255, 255, 0.2),
    -1.375em -0.125em 0 ${theme.dropsColor};
}
`;

const Drop = styled.div`
  &:after {
    content: '';
    position: absolute;
    z-index: 2;
    top: 50%;
    left: 50%;
    width: 1.125em;
    height: 1.125em;
    margin: -1em 0 0 -0.25em;
    background: ${({ theme }) => theme.dropsColor};
    border-radius: 100% 0 60% 50% / 60% 0 100% 50%;
    box-shadow: 0.625em 0.875em 0 -0.125em rgba(255, 255, 255, 0.2),
      -0.875em 1.125em 0 -0.125em rgba(255, 255, 255, 0.2),
      -1.375em -0.125em 0 rgba(255, 255, 255, 0.2);
    transform: rotate(-28deg);
    animation: ${DropAnimation} 3s linear infinite;
  }
`;

Drop.propTypes = {
  theme: ThemePropType,
};

Drop.defaultProps = {
  theme: defaultTheme,
};

Drop.displayName = 'Drop';

export default Drop;
