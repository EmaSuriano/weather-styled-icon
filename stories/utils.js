import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { WeatherThemeProvider } from '../src';
import { ThemePropType } from '../src/constants';

export const CenterContainer = styled.div`
  ${props =>
    props.theme.backgroundColor &&
    `background-color: ${props.theme.backgroundColor}`};
  text-align: center;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const BackgroundContainer = styled(CenterContainer)`
  background-color: ${props => props.color};
`;

export const ThemeContainer = ({ theme, children }) => (
  <WeatherThemeProvider theme={theme}>
    <BackgroundContainer>{children}</BackgroundContainer>
  </WeatherThemeProvider>
);

ThemeContainer.propTypes = {
  theme: ThemePropType.isRequired,
  children: PropTypes.node,
};

ThemeContainer.defaultProps = {
  children: PropTypes.node.isRequired,
};
