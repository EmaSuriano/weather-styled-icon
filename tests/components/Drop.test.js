import React from 'react';
import Drop from '../../src/components/Drop';
import { renderWithTheme } from '../utils';

describe('<Drop />', () => {
  it('should should render div with theme', () => {
    const wrapper = renderWithTheme(<Drop />);

    expect(wrapper).toMatchSnapshot();
  });

  xit('DropAnimation should return keyframes with theme', () => {
    // TOOD: Find a proper way to test Animations!
  });
});
