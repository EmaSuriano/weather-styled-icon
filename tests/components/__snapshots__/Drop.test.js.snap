// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drop /> should should render div with theme 1`] = `
.c0:after {
  content: '';
  position: absolute;
  z-index: 2;
  top: 50%;
  left: 50%;
  width: 1.125em;
  height: 1.125em;
  margin: -1em 0 0 -0.25em;
  background: blue;
  border-radius: 100% 0 60% 50% / 60% 0 100% 50%;
  box-shadow: 0.625em 0.875em 0 -0.125em rgba(255,255,255,0.2), -0.875em 1.125em 0 -0.125em rgba(255,255,255,0.2), -1.375em -0.125em 0 rgba(255,255,255,0.2);
  -webkit-transform: rotate(-28deg);
  -ms-transform: rotate(-28deg);
  transform: rotate(-28deg);
  -webkit-animation: lnNETy 3s linear infinite;
  animation: lnNETy 3s linear infinite;
}

<div
  className="c0"
/>
`;
